core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -37,6 +37,41 @@
   private static final int TEMPORARY_SUFFIX_LENGTH =
       String.format(TEMPORARY_SUFFIX_FORMAT, 0).length();
 
+  /**
+   * Returns the basename of the path, consistent with the Unix basename command.
+   *
+   * Examples:
+   *
+   * <pre>
+   *   basename("/") = "/"
+   *   basename("/abc") = "abc"
+   *   basename("///") = "/"
+   *   basename("abc/" = "abc"
+   *   basename("") = ""
+   *   basename("/a.txt") = "a.txt"
```

Comment:
can you provide nested examples, like /a/b/c?

Reply:
done